### PR TITLE
Issue #3058403 by mtodor, daniel.bosen: Build Thunder performance test pipeline

### PR DESCRIPTION
**Pull request by mtodor**
Make sure these boxes are checked before submitting your pull request - thank you!

- [ ] All coding styles are fulfilled. ([How to check for cs issues?](https://github.com/BurdaMagazinOrg/thunder-dev-tools/blob/master/README.md#code-style-guidelines))
- [ ] All tests are running locally. ([How to run the test?](https://github.com/thunder/thunder-distribution/blob/8.x-3.x/docs/development.md#how-to-run-the-tests))
- [ ] Necessary update hooks are provided.
- [ ] User roles have correct access for new introduced permission.
- [ ] Every thunder module has a README.md in its root. Follow [this guidelines](https://www.drupal.org/node/2181737), but we don't need every topic.
- [ ] Code is covered with well-balanced amount of inline comments.

If you are really awesome, then your feature is covered by additional tests. Well done!

### CODE REVIEW

**Review by dbosen — CHANGES_REQUESTED** [review]
Review of Lambda:

**Lines 19 - 44:**

Can those parameter checks also be done in the api gateway? no reason to call the lambda, if the parameters are already messed up.

**Line 48:**

`
            taskDefinition=event['queryStringParameters']['task_defintion_template']
`

use variable task_defintion_template (declared in line 18) instead

**Line 54:**

`
        task_definition = task_definition['taskDefinition']
`

Reusing a variable name is a bit confusing. Why not do something like the following on line 47 instead?

`
        task_definition = ecs_client.describe_task_definition(
            taskDefinition=event['queryStringParameters']['task_defintion_template']
        )['taskDefinition']
`

**Line 72:**
`
    # Remove automatically added properties
    task_definition.pop('taskDefinitionArn', None)
    task_definition.pop('status', None)
    task_definition.pop('requiresAttributes', None)
    task_definition.pop('compatibilities', None)
    task_definition.pop('revision', None)
`

why remove those keys, if we assign the needed values one by one in the registration part (see line 106 and the following ones)?

File: scripts/travis/08-build-docker-performance-image.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+#
+# Download stored project artifact on S3 and pack it into Thunder performance docker image
+
+# Get Thunder performance Docker project for packaging
+THUNDER_PERFORMANCE="${TEST_DIR}/../docker-thunder-performance"
+git clone https://github.com/thunder/docker-thunder-performance.git "${THUNDER_PERFORMANCE}"
+
+# Download project artifact from S3
+AWS_ACCESS_KEY_ID="${ARTIFACTS_KEY}" AWS_SECRET_ACCESS_KEY="${ARTIFACTS_SECRET}" aws s3 cp "s3://thunder-builds/${PROJECT_ARTIFACT_FILE_NAME}" "${PROJECT_ARTIFACT_FILE}"
```
Comment (dbosen) [inline scripts/travis/08-build-docker-performance-image.sh]: This is being done in several places now, we should provide the AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY globally in the 01-setup-environment.sh file

File: tests/src/Nightwatch/Commands/paragraphs/add.js [hunk 2]
```diff
@@ -28,7 +28,6 @@ exports.command = function add(fieldName, type, position) {
     .scrollIntoMiddleOfView(
       `//table[contains(@id, "${fieldNameId}-values")]/tbody/tr[${addButtonPosition}]//input`
     )
-    .pause(10000)
```
Comment (dbosen) [inline tests/src/Nightwatch/Commands/paragraphs/add.js:31]: Amazing performance improvement right here...

**Review by dbosen — CHANGES_REQUESTED** [review]

File: tests/src/Nightwatch/Commands/paragraphs/add.js (hunk 2 above)
Comment (mtodor) [reply tests/src/Nightwatch/Commands/paragraphs/add.js:31]: `4x` improvement. I doubt we will ever beat that.

File: scripts/travis/08-build-docker-performance-image.sh (hunk 1 above)
Comment (mtodor) [reply scripts/travis/08-build-docker-performance-image.sh]: 👍

**Review by dbosen — APPROVED** [review]

**Pull request merged**